### PR TITLE
 Add print of numbers to the left of the diff

### PR DESCRIPTION
**Pull request by thommyj**
Add input numbers to the left of the binary string. Also add an readme with example

### CODE REVIEW

File: diff64.c [hunk 1]
```diff
@@ -77,12 +77,37 @@ static uint32_t get_bit(const uint64_t number, const uint32_t index)
     return ((number >> index) & 1uLL);
 }
 
+static uint32_t get_max_number_length(const uint64_t left_number, const uint64_t right_number)
```
Comment (gaspar-kolumban) [inline diff64.c]: Add this to the declaration section

File: diff64.c [hunk 2]
```diff
@@ -77,12 +77,37 @@ static uint32_t get_bit(const uint64_t number, const uint32_t index)
...
+{
+    uint32_t left_number_length;
+    uint32_t right_number_length;
+    uint32_t max_number_length;
+    const uint32_t max_string_length = MAX_COMPARE_LENGTH/4 + 1;
+    char number_length[max_string_length];
```
Comment (gaspar-kolumban) [inline diff64.c]: No VLA's please. Move the value of max_string_length inside the brackets and use "u" for the digits.

File: diff64.c [hunk 3]
```diff
@@ -97,9 +122,15 @@ static void differentiate(const uint64_t left_number, const uint64_t right_numbe
         snprintf(&(lower_result[result_index]),  write_size, "%*u", ELEMENT_LENGTH, right_bit);
     }
 
+    max_number_length =  get_max_number_length(left_number, right_number);
```
Comment (gaspar-kolumban) [inline diff64.c]: Add this line to the declaration of max_number_length and make it const (also there is two spaces before the function call :P)

File: diff64.c [hunk 4]
```diff
@@ -97,9 +122,15 @@ static void differentiate(const uint64_t left_number, const uint64_t right_numbe
...
+
+    printf("  %*c ", max_number_length, ' ');
     printf("%s\n", legend_result);
+    printf("0x%0*lX ", max_number_length, left_number);
```
Comment (gaspar-kolumban) [inline diff64.c]: I prefer lower case "x".

File: diff64.c [hunk 5]
```diff
@@ -97,9 +122,15 @@ static void differentiate(const uint64_t left_number, const uint64_t right_numbe
...
     printf("%s\n", upper_result);
+    printf("  %*c ", max_number_length, ' ');
     printf("%s\n", middle_result);
+    printf("0x%0*lX ", max_number_length, right_number);
```
Comment (gaspar-kolumban) [inline diff64.c]: I prefer lower case "x".

File: diff64.c [hunk 6]
```diff
@@ -77,12 +77,37 @@ static uint32_t get_bit(const uint64_t number, const uint32_t index)
...
+    left_number_length = strlen(number_length);
+    snprintf(number_length, max_string_length, "%lX", right_number);
+    right_number_length = strlen(number_length);
+    if (right_number_length > left_number_length)
+    {
+        max_number_length = right_number_length;
+    }
+    else
+    {
+        max_number_length = right_number_length;
```
Comment (gaspar-kolumban) [inline diff64.c]: Should be left_number_length for the else branch.

File: diff64.c [hunk 7]
```diff
@@ -77,12 +77,37 @@ static uint32_t get_bit(const uint64_t number, const uint32_t index)
...
 
+static uint32_t get_max_number_length(const uint64_t left_number, const uint64_t right_number)
+{
+    uint32_t left_number_length;
+    uint32_t right_number_length;
+    uint32_t max_number_length;
+    const uint32_t max_string_length = MAX_COMPARE_LENGTH/4 + 1;
+    char number_length[max_string_length];
+
+    snprintf(number_length, max_string_length, "%lX", left_number);
```
Comment (gaspar-kolumban) [inline diff64.c]: I prefer lower case "x", not that it really matters here but for the sake of consistency, if nothing else =)

File: diff64.c [hunk 8]
```diff
@@ -77,12 +77,37 @@ static uint32_t get_bit(const uint64_t number, const uint32_t index)
...
+    left_number_length = strlen(number_length);
+    snprintf(number_length, max_string_length, "%lX", right_number);
```
Comment (gaspar-kolumban) [inline diff64.c]: I prefer lower case "x", not that it really matters here but for the sake of consistency, if nothing else =)

**Review by gaspar-kolumban — CHANGES_REQUESTED** [review]

**Comment by thommyj** [comment]
fixing in another pull request

**Pull request closed without merging**